dockstore-webservice/src/main/java/io/dockstore/webservice/resources/AbstractWorkflowResource.java

```diff
@@ -256,136 +215,243 @@ protected void updateDBWorkflowWithSourceControlWorkflow(Workflow workflow, Work
         }
     }
 
-
-    protected T upsertVersion(String repository, String username, String gitReference, String installationId, WorkflowMode workflowMode) {
-        // Retrieve the user who triggered the call (may not exist on Dockstore)
-        User sendingUser = GitHubHelper.findUserByGitHubUsername(this.tokenDAO, this.userDAO, username, false);
+    /**
+     * Handle webhooks from GitHub apps (redirected from AWS Lambda)
+     * - Create services and workflows when necessary
+     * - Add or update version for corresponding service and workflow
+     * @param repository Repository path (ex. dockstore/dockstore-ui2)
+     * @param username Username of GitHub user that triggered action
+     * @param gitReference Tag reference from GitHub (ex. 1.0)
+     * @param installationId GitHub App installation ID
+     * @return List of new and updated workflows
+     */
+    protected List<Workflow> githubWebhookRelease(String repository, String username, String gitReference, String installationId) {
+        // Retrieve the user who triggered the call (must exist on Dockstore if workflow is not already present)
+        User user = GitHubHelper.findUserByGitHubUsername(this.tokenDAO, this.userDAO, username, false);
 
         // Get Installation Access Token
         String installationAccessToken = gitHubAppSetup(installationId);
 
-        // Call common upsert code
-        String dockstoreServicePath = upsertVersionHelper(repository, gitReference, null, workflowMode, installationAccessToken);
+        // Grab Dockstore YML from GitHub
+        GitHubSourceCodeRepo gitHubSourceCodeRepo = (GitHubSourceCodeRepo)SourceCodeRepoFactory.createGitHubAppRepo(installationAccessToken);
+        SourceFile dockstoreYml = gitHubSourceCodeRepo.getDockstoreYml(repository, gitReference);
 
-        // Add user to service if necessary
-        T entity = workflowDAO.findByPath(dockstoreServicePath, false, entityClass).get();
-        if (sendingUser != null && !entity.getUsers().contains(sendingUser)) {
-            entity.getUsers().add(sendingUser);
-        }
+        // Parse Dockstore YML and perform appropriate actions
+        Yaml yaml = new Yaml();
+        try {
+            Map<String, Object> map = yaml.load(dockstoreYml.getContent());
+            double versionString = (double)map.get("version");
+
+            if (Objects.equals(version11, versionString)) {
+                // 1.1 - Only works with services
```

Comment:
Please write a test with a 1.1 and 1.2 schema file checked-in. I think we can make some quick decisions to make 1.2 support both and be compatible (i.e. can switch the version number from 1.1 to 1.2 to "upgrade", a 1.2 file can be read as a 1.1 file since your new keys should just be ignored)

Reply:
I do have tests for schema 1.1 for services and 1.2 for workflows. Do you mean have one a test that has both 1.1 and 1.2 services? I was hoping that the 1.2 services could be added in the next PR since I didn't want this one getting too big.

Reply:
https://github.com/dockstore/dockstore/issues/3237